Libraries/src/Amazon.Lambda.DynamoDBEvents/ExtensionMethods.cs

```diff
@@ -0,0 +1,163 @@
+        }
+
+        /// <summary>
+        /// Writes a single DynamoDB attribute as a json. May be called recursively for maps.
+        /// </summary>
+        /// <param name="writer">JSON writer</param>
+        /// <param name="item">Dictionary representing a DynamoDB item, or a map within an item</param>
+        private static void WriteJson(Utf8JsonWriter writer, Dictionary<string, AttributeValue> item)
+        {
+            writer.WriteStartObject();
+
+            foreach (var attribute in item)
+            {
+                writer.WritePropertyName(attribute.Key);
+                WriteJsonValue(writer, attribute.Value);
+            }
+
+            writer.WriteEndObject();
+        }
+
+        /// <summary>
+        /// Writes a single DynamoDB attribute value as a json value
+        /// </summary>
+        /// <param name="writer">JSON writer</param>
+        /// <param name="attribute">DynamoDB attribute</param>
+        private static void WriteJsonValue(Utf8JsonWriter writer, AttributeValue attribute)
+        {
+            if (attribute.S != null)
+            {
+                writer.WriteStringValue(attribute.S);
+            }
+            else if (attribute.N != null)
+            {
+#if NETCOREAPP3_1  // WriteRawValue was added in .NET 6, but we need to write out Number values without quotes
+                using (var document = JsonDocument.Parse(attribute.N))
+                {
+                    document.WriteTo(writer);
+                }
+#else
+                writer.WriteRawValue(attribute.N);
+#endif
+            }
+            else if (attribute.B != null)
+            {
+                writer.WriteBase64StringValue(attribute.B.ToArray());
+            }
+            else if (attribute.BOOL != null)
+            {
+                writer.WriteBooleanValue(attribute.BOOL.Value);
+            }
+            else if (attribute.NULL != null)
+            {
+                writer.WriteNullValue();
+            }
+            else if (attribute.M != null)
+            {
+                WriteJson(writer, attribute.M);
+            }
+            else if (attribute.L != null)
+            {
+                writer.WriteStartArray();
+                foreach (var item in attribute.L)
+                {
+                    WriteJsonValue(writer, item);
+                }
+                writer.WriteEndArray();
+            }
+            else if (attribute.SS != null)
+            {
+                writer.WriteStartArray();
+                foreach (var item in attribute.SS)
+                {
+                    writer.WriteStringValue(item);
+                }
+                writer.WriteEndArray();
+            }
+            else if (attribute.NS != null)
+            {
+                writer.WriteStartArray();
+                foreach (var item in attribute.NS)
+                {
+#if NETCOREAPP3_1  // WriteRawValue was added in .NET 6, but we need to write out Number values without quotes
```

Comment:
In other places where we check for .NET 6 or later we use: `#if NET6_0_OR_GREATER`. Is there a reason you didn't use the same pattern here?

Reply:
My thinking was when we drop 3.1 support, we'd be more likely to search for `NETCOREAPP3_1` during cleanup.